osu.Game/Overlays/AccountCreation/ScreenWarning.cs

```diff
@@ -26,11 +26,14 @@ public class ScreenWarning : AccountCreationScreen
         [Resolved(CanBeNull = true)]
         private IAPIProvider api { get; set; }
 
+        [Resolved(canBeNull: true)]
+        private OsuGame game { get; set; }
+
         private const string help_centre_url = "/help/wiki/Help_Centre#login";
 
         public override void OnEntering(IScreen last)
         {
-            if (string.IsNullOrEmpty(api?.ProvidedUsername))
+            if (string.IsNullOrEmpty(api?.ProvidedUsername) || game.UseDevelopmentServer)
```

Comment:
If nullability of `game` is to be allowed (which it wasn't before), then this should be

```suggestion
            if (string.IsNullOrEmpty(api?.ProvidedUsername) || game?.UseDevelopmentServer)
```

lest it crash on a `null`.